components/input/Input.js

```diff
@@ -0,0 +1,161 @@
+import React, { Component, createElement } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { IconChevronUpSmallOutline, IconChevronDownSmallOutline, IconErrorSmallOutline } from '@teamleader/ui-icons';
+import InputMetaPropTypes from './InputMetaPropTypes';
+import Counter from '../counter';
+import theme from './theme.css';
+
+export default class Input extends Component {
+  static propTypes = {
+    bold: PropTypes.bool,
+    className: PropTypes.string,
+    counter: PropTypes.number,
+    disabled: PropTypes.bool,
+    icon: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
+    iconPlacement: PropTypes.oneOf(['left', 'right']),
+    id: PropTypes.string,
+    meta: InputMetaPropTypes,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
+    placeholder: PropTypes.string,
+    readOnly: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'medium', 'large']),
+    step: PropTypes.number,
+    type: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  };
+
+  static defaultProps = {
+    iconPlacement: 'left',
+    disabled: false,
+    placeholder: '',
+    readOnly: false,
+    size: 'medium',
+    step: 1,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.isNumberInput = Boolean(props.type === 'number');
+    this.handleChange = this.onChange.bind(this);
+    this.handleIncreaseValue = this.increaseValue.bind(this);
+    this.handleDecreaseValue = this.decreaseValue.bind(this);
```

Comment:
We are using the `stage-0` preset, feel free to use new function bind operator `::`.

Reply:
🎉